### PR TITLE
Privacy experiences initial pages

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fides/issues/3122, closes https://github.com/ethyca/fides/issues/3123

### Code Changes

* [x] Update TS autogenerated types
* [x] Initial page for privacy experiences (table view, edit view)
* [x] Table for privacy experiences (easy to do thanks to @TheAndrewJackson 's table refactor!)
* [x] Tweaks to side nav that came up now that we have multiple child side navs
  * [x] Consent page now has its own "page" which just redirects to notices. This makes the "tree" view of the side nav work properly (before I just had the Consent page linking to "/consent/privacy-notices" but that makes "/consent/" not highlight when under "/consent/privacy-experiences" since it doesn't look like part of the same tree)
* [x] Cypress tests

### Steps to Confirm

This is a little tricky to test right now since the backend isn't in `main` yet, so we can't just use fidesplus double edge. I've found the easiest thing to do is to run the OSS fides backend, then in `features.slice.ts`, set `plus=true` instead of the plus health check.

You'll also need a privacy experiences json file to load in, similar to how we load in privacy notices while there aren't any out of the box ones yet.

<details>
<summary>privacyExperiences.json</summary>

```json
[
  {
    "component": "overlay",
    "regions": ["eu_fr", "eu_ie"],
    "delivery_mechanism": "banner"
  },
  {
    "component": "privacy_center",
    "regions": ["us_ca"],
    "delivery_mechanism": "link"
  },
  {
    "component": "overlay",
    "regions": ["eu_de"],
    "delivery_mechanism": "link"
  }
]

```

</details>

* Click the "Privacy requests" nav link, then in the side nav, click "Privacy experiences"
* If you've loaded the experiences above, you should see them in the table
* You should be able to click into a row and go to a work in progress page

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### Description Of Changes

https://user-images.githubusercontent.com/24641006/235742903-870975e0-c3f9-4907-86ff-6ce28ecc5a8c.mov

A few things about this table:
image

* We're punting on how the locations should show up for now. Still need to think through how we will group them and display them
* The privacy experience table says "Last update" while the notice table says "Last modified". Should we make these consistent @mfbrown ? Simon [deferred to you](https://www.figma.com/file/dmEwdK3xZwjKfGVQ9t66xe?node-id=5292:164434#431038877) :) 
* ~~The enable/disable button doesn't currently work with the backend yet due to the [backend doing more validation than necessary](https://ethyca.slack.com/archives/C04S7K0T2V7/p1683047701741599?thread_ts=1683047447.874469&cid=C04S7K0T2V7) right now. will be fixed soon!~~ Fixed!

image

* I copied the empty state from notices. I feel like the copy still applies since data uses drive everything, but it might read weird, not sure. Let me know if you want to change this copy @mfbrown

### CODE REVIEW

File: clients/admin-ui/src/features/common/table/FidesTable.tsx [hunk 1]
```diff
@@ -133,7 +134,7 @@ export const FidesTable = <T extends FidesObject>({
                     ? { backgroundColor: "gray.50", cursor: "pointer" }
                     : undefined
                 }
-                data-testid={`row-${rowName}`}
+                data-testid={`row-${rowName ?? rowId}`}
```
Comment (allisonking) [inline clients/admin-ui/src/features/common/table/FidesTable.tsx:137]: privacy experiences don't have a `name` field, so rely on `id` for the test id instead

File: clients/admin-ui/src/features/privacy-experience/privacy-experience.slice.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+} from "~/types/api";
+
+export interface State {
+  page?: number;
+  pageSize?: number;
+}
+
+const initialState: State = {
+  page: 1,
+  pageSize: 10,
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/privacy-experience/privacy-experience.slice.ts:19]: Should the pagesize be a small number like 10 when there is no way to switch between pages?

**Review by TheAndrewJackson — APPROVED** [review]
Looks great! Just one small nit about the pageSize

File: clients/admin-ui/src/features/privacy-experience/privacy-experience.slice.ts (hunk 2 above)
Comment (allisonking) [reply clients/admin-ui/src/features/privacy-experience/privacy-experience.slice.ts:19]: good idea, I'll increase to 50 😎

**Pull request merged**